src/ansys/fluent/core/utils/search.py

```diff
@@ -212,10 +238,280 @@ def inner(tree, path, root_path):
+    match_whole_word: Optional[bool] = False,
+    match_case: Optional[bool] = False,
+):
+    """Search for a word through the Fluent's object hierarchy.
+
+    Parameters
+    ----------
+    search_string: str
+        The word to search for. Semantic search is default.
+    language: str
+        The language for the semantic search.
+        English is default for the semantic search.
+        ISO 639-3 code of the language to be used for semantic search.
+        See `https://omwn.org/omw1.html` for the list of supported languages.
+        The default value is `eng` for English language.
+    wildcard: bool
+        Whether to use wildcard pattern. If ``True`` will match wildcard pattern based on ``fnmatch`` module and
+        will turn off semantic matching.
+    match_whole_word: bool
+        Whether to get exact match. If ``True`` will match exact string and will turn off semantic matching.
+    match_case: bool
+        Whether to match case. If ``True`` will match case-insensitive case.
+
+    Raises
+    ------
+    ValueError
+        If both ``wildcard`` and ``match_whole_word`` are ``True``.
+
+    Examples
+    --------
+    >>> import ansys.fluent.core as pyfluent
+    >>> pyfluent.search("font", match_whole_word=True)
+    >>> pyfluent.search("Font")
+    >>> pyfluent.search("iter*", wildcard=True)
+    >>> pyfluent.search("读", language="cmn")   # search 'read' in Chinese
+    The most similar API objects are:
+    <solver_session>.file.read (Command)
+    <solver_session>.file.import_.read (Command)
+    <solver_session>.mesh.surface_mesh.read (Command)
+    <solver_session>.tui.display.display_states.read (Command)
+    <meshing_session>.tui.display.display_states.read (Command)
+    """
+    if wildcard and match_whole_word:
+        raise ValueError(
+            "``wildcard`` cannot be ``True`` when ``match_whole_word`` is ``True``."
+        )
+    elif language and match_whole_word:
+        warnings.warn(
+            "``match_whole_word=True`` will match exact string and will turn off semantic matching.",
+            UserWarning,
+        )
+    elif match_whole_word:
+        warnings.warn(
+            "``match_whole_word=True`` will turn off wildcard matching.",
+            UserWarning,
+        )
+
+    api_object_names = get_api_tree_file_name(name=True)
+    names = [
+        line.rstrip("\n") for line in open(api_object_names, "r", encoding="utf-8")
+    ]
+
+    if wildcard:
+        queries = _process_wildcards(search_string, names)
+    elif match_whole_word:
+        queries = _process_misspelled(
+            word=search_string,
+            names=names,
+            match_whole_word=match_whole_word,
+        )
+        if match_case:
+            queries = _process_misspelled(
+                word=search_string,
+                names=names,
+                match_case=match_case,
+            )
+    elif not wildcard and not match_whole_word and not match_case:
+        if _download_nltk_data():
```

Comment:
Instead of checking for the downloaded data every time, how about the following approach?
```
try:
    _search_semantic(...)
except LookupError:
    _download_nltk_data()
    _search_semantic(...)
```

Reply:
Yes, this looks good. Thank you.

Reply:
Done. Thank you.